spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/EnableAutoConfigurationImportSelector.java

```diff
@@ -205,6 +206,22 @@ private MetadataReaderFactory getMetadataReaderFactory() {
 		}
 	}
 
+	private void checkExcludedClasses(List<String> configurations,
+			Set<String> exclusions) {
+		StringBuilder message = new StringBuilder();
+		for (String exclusion : exclusions) {
+			if (!configurations.contains(exclusion)) {
+				message.append("\t- ").append(exclusion).append("\n");
+			}
+		}
+		if (!message.toString().isEmpty()) {
+			throw new IllegalStateException(
+					"Could not exclude the following classes from autoconfiguration:\n"
```

Comment:
How about throwing a dedicated exception for this so that we can have a `FailureAnalyzer` that goes with it. Maybe going a tad too far ...

If we don't, I think the error message should state why they couldn't be excluded (because these are not  auto-configurations or something).